plan/gen_physical_plans.go

```diff
@@ -416,8 +457,12 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *requiredProp) ([]PhysicalPlan, boo
 // If the hint is not figured, we will pick all candidates.
 func (p *LogicalJoin) exhaustPhysicalPlans(prop *requiredProp) []PhysicalPlan {
 	mergeJoins := p.getMergeJoin(prop)
-	if (p.preferJoinType&preferMergeJoin) > 0 && len(mergeJoins) > 0 {
-		return mergeJoins
+	if (p.preferJoinType & preferMergeJoin) > 0 {
```

Comment:
Can we put this code logic into function `p.getMergeJoin`?

Reply:
Ummm, What's your means? Put the `p.getEnforcedMergeJoin` into `p.getMergeJoin`?

Reply:
Yes, maybe put line 460~465 to function `getMergeJoin` is better.